scene/gui/graph_element.cpp

```diff
@@ -0,0 +1,244 @@
+	switch (p_what) {
+		case NOTIFICATION_SORT_CHILDREN: {
+			_resort();
+		} break;
+
+		case NOTIFICATION_LAYOUT_DIRECTION_CHANGED:
+		case NOTIFICATION_TRANSLATION_CHANGED:
+		case NOTIFICATION_THEME_CHANGED: {
+			update_minimum_size();
+			queue_redraw();
+		} break;
+	}
+}
+
+void GraphElement::_validate_property(PropertyInfo &p_property) const {
+	Control::_validate_property(p_property);
+	GraphEdit *graph = Object::cast_to<GraphEdit>(get_parent());
+	if (graph) {
+		if (p_property.name == "position") {
+			p_property.usage |= PROPERTY_USAGE_READ_ONLY;
+		}
+	}
+}
+
+void GraphElement::set_position_offset(const Vector2 &p_offset) {
+	position_offset = p_offset;
+	emit_signal(SNAME("position_offset_changed"));
+	queue_redraw();
+}
+
+Vector2 GraphElement::get_position_offset() const {
+	return position_offset;
+}
+
+void GraphElement::set_selected(bool p_selected) {
+	if (!is_selectable() || selected == p_selected) {
+		return;
+	}
+	selected = p_selected;
+	emit_signal(p_selected ? SNAME("node_selected") : SNAME("node_deselected"));
+	queue_redraw();
+}
+
+bool GraphElement::is_selected() {
+	return selected;
+}
+
+void GraphElement::set_drag(bool p_drag) {
+	if (p_drag) {
+		drag_from = get_position_offset();
+	} else {
+		emit_signal(SNAME("dragged"), drag_from, get_position_offset()); // Required for undo/redo.
+	}
+}
+
+Vector2 GraphElement::get_drag_from() {
+	return drag_from;
+}
+
+void GraphElement::gui_input(const Ref<InputEvent> &p_ev) {
+	ERR_FAIL_COND(p_ev.is_null());
+
+	Ref<InputEventMouseButton> mb = p_ev;
+	if (mb.is_valid()) {
+		ERR_FAIL_COND_MSG(get_parent_control() == nullptr, "GraphElement must be the child of a GraphEdit node.");
+
+		if (mb->is_pressed() && mb->get_button_index() == MouseButton::LEFT) {
+			Vector2 mpos = mb->get_position();
+
+			Ref<Texture2D> resizer = get_theme_icon(SNAME("resizer"));
+
+			if (resizable && mpos.x > get_size().x - resizer->get_width() && mpos.y > get_size().y - resizer->get_height()) {
+				resizing = true;
+				resizing_from = mpos;
+				resizing_from_size = get_size();
+				accept_event();
+				return;
+			}
+
+			emit_signal(SNAME("raise_request"));
+		}
+
+		if (!mb->is_pressed() && mb->get_button_index() == MouseButton::LEFT) {
+			resizing = false;
+		}
+	}
+
+	Ref<InputEventMouseMotion> mm = p_ev;
+	if (resizing && mm.is_valid()) {
+		Vector2 mpos = mm->get_position();
+
+		Vector2 diff = mpos - resizing_from;
+
+		emit_signal(SNAME("resize_request"), resizing_from_size + diff);
+	}
+}
+
+void GraphElement::set_resizable(bool p_enable) {
+	resizable = p_enable;
+	queue_redraw();
+}
+
+bool GraphElement::is_resizable() const {
+	return resizable;
+}
+
+void GraphElement::set_draggable(bool p_draggable) {
+	draggable = p_draggable;
+}
+
+bool GraphElement::is_draggable() {
+	return draggable;
+}
+
+void GraphElement::set_selectable(bool p_selectable) {
+	if (!p_selectable) {
+		set_selected(false);
+	}
+	selectable = p_selectable;
+}
```

Comment:
A similar check can probably also be added here.

Reply:
There is already a check in set_selected and this method has no additional side effects, so I wouldn't add another check here.